### PR TITLE
MAINT-52254: Fix wrong gamification points when comment an activity

### PR DESCRIPTION
**Pull request by hakermi**
**ISSUE**: When comment an activity both gamification for comment and receive comment goes to the commenter and missing comment plugin for space stream activities
**FIX**: Correct the wrong matching between receiver and comment sender and adding new plugin event for space activities

### CODE REVIEW

File: services/src/main/java/org/exoplatform/addons/gamification/GamificationConstant.java [hunk 1]
```diff
@@ -35,8 +35,10 @@
     String GAMIFICATION_SOCIAL_COMMENT_NETWORK_STREAM                                = "commentOnNetworkStream";
     /**  COMMENT ON SPACE CONTEXT */
     String GAMIFICATION_SOCIAL_COMMENT_SPACE_STREAM                                  = "commentOnSpaceStream";
-    /**  COMMENT BY A USER */
-    String GAMIFICATION_SOCIAL_COMMENT_ADD                                           = "addComment";
+    /**  COMMENT BY A USER ON NETWORK STREAM*/
+    String GAMIFICATION_SOCIAL_COMMENT_ADD_NETWORK_STREAM                            = "addComment";
+    /**  COMMENT BY A USER ON SPACE STREAM*/
+    String GAMIFICATION_SOCIAL_COMMENT_ADD_SPACE_STREAM                              = "addCommentOnSpaceStream";
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/addons/gamification/GamificationConstant.java]: We have here 4 events : commentOnNetworkStream, commentOnSpaceStream, addComment, addCommentOnSpaceStream.
Can you change event name value to have something more readable ?
We should have : receiveCommentOnSpaceStream, receiveCommentonNetworkStream, addCommentOnSpaceStream, receiveCommentOnSpaceStream

File: extension/src/main/webapp/WEB-INF/conf/gamification/storage/rule/rules-storage-configuration.xml [hunk 2]
```diff
@@ -299,7 +299,7 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
     <external-component-plugins>
         <target-component>org.exoplatform.addons.gamification.service.setting.rule.RuleRegistry</target-component>
         <component-plugin>
-            <name>rule.AddCommenOnSpaceStream</name>
+            <name>rule.receiveCommentOnSpaceStream</name>
```
Comment (rdenarie) [inline extension/src/main/webapp/WEB-INF/conf/gamification/storage/rule/rules-storage-configuration.xml:302]: Why do we need 2 differents rules for receive comment ? Should we have only one ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/addons/gamification/GamificationConstant.java (hunk 1 above)
Comment (hakermi) [reply services/src/main/java/org/exoplatform/addons/gamification/GamificationConstant.java]: Okay done

File: extension/src/main/webapp/WEB-INF/conf/gamification/storage/rule/rules-storage-configuration.xml (hunk 2 above)
Comment (hakermi) [reply extension/src/main/webapp/WEB-INF/conf/gamification/storage/rule/rules-storage-configuration.xml:302]: one for space stream and one for the network stream

Comment (rdenarie) [reply extension/src/main/webapp/WEB-INF/conf/gamification/storage/rule/rules-storage-configuration.xml:302]: Is it possible to have one rule for both or not ? I have the feeling that we duplicate something

Comment (hakermi) [reply extension/src/main/webapp/WEB-INF/conf/gamification/storage/rule/rules-storage-configuration.xml:302]: We need two to have two different messages the keys are created dynamically with by concatenating rule title and the event name and the plugin name is for more consistency and readability :
Activity Stream (space) : Receive comment
Activity Stream (network) : Receive comment

all the plugins are created with this way

File: services/src/main/java/org/exoplatform/addons/gamification/GamificationConstant.java [hunk 3]
```diff
@@ -32,11 +32,13 @@
...
     /**  COMMENT ON USER CONTEXT */
-    String GAMIFICATION_SOCIAL_COMMENT_NETWORK_STREAM                                = "commentOnNetworkStream";
+    String GAMIFICATION_SOCIAL_COMMENT_NETWORK_STREAM                                = "receiveCommentOnNetworkStream";
     /**  COMMENT ON SPACE CONTEXT */
-    String GAMIFICATION_SOCIAL_COMMENT_SPACE_STREAM                                  = "commentOnSpaceStream";
-    /**  COMMENT BY A USER */
-    String GAMIFICATION_SOCIAL_COMMENT_ADD                                           = "addComment";
+    String GAMIFICATION_SOCIAL_COMMENT_SPACE_STREAM                                  = "receiveCommentOnSpaceStream";
+    /**  COMMENT BY A USER ON NETWORK STREAM*/
+    String GAMIFICATION_SOCIAL_COMMENT_ADD_NETWORK_STREAM                            = "addCommentOnNetworkStream";
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/addons/gamification/GamificationConstant.java]:
```suggestion
    String GAMIFICATION_SOCIAL_ADD_COMMENT_NETWORK_STREAM                            = "addCommentOnNetworkStream";
```

File: services/src/main/java/org/exoplatform/addons/gamification/GamificationConstant.java [hunk 4]
```diff
@@ -32,11 +32,13 @@
...
+    /**  COMMENT BY A USER ON SPACE STREAM*/
+    String GAMIFICATION_SOCIAL_COMMENT_ADD_SPACE_STREAM                              = "addCommentOnSpaceStream";
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/addons/gamification/GamificationConstant.java]:
```suggestion
    String GAMIFICATION_SOCIAL_ADD_COMMENT_SPACE_STREAM                              = "addCommentOnSpaceStream";
```

File: services/src/main/java/org/exoplatform/addons/gamification/GamificationConstant.java [hunk 5]
```diff
@@ -32,11 +32,13 @@
     /*************************************************** CONSTANTS TO MANAGE COMMENT GAMIFICATION PROCESS **********************************************/
     /***************************************************************************************************************************************************/
     /**  COMMENT ON USER CONTEXT */
-    String GAMIFICATION_SOCIAL_COMMENT_NETWORK_STREAM                                = "commentOnNetworkStream";
+    String GAMIFICATION_SOCIAL_COMMENT_NETWORK_STREAM                                = "receiveCommentOnNetworkStream";
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/addons/gamification/GamificationConstant.java]:
```suggestion
    String GAMIFICATION_SOCIAL_RECEIVE_COMMENT_NETWORK_STREAM                                = "receiveCommentOnNetworkStream";
```

File: services/src/main/java/org/exoplatform/addons/gamification/GamificationConstant.java [hunk 6]
```diff
@@ -32,11 +32,13 @@
...
     /**  COMMENT ON SPACE CONTEXT */
-    String GAMIFICATION_SOCIAL_COMMENT_SPACE_STREAM                                  = "commentOnSpaceStream";
-    /**  COMMENT BY A USER */
-    String GAMIFICATION_SOCIAL_COMMENT_ADD                                           = "addComment";
+    String GAMIFICATION_SOCIAL_COMMENT_SPACE_STREAM                                  = "receiveCommentOnSpaceStream";
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/addons/gamification/GamificationConstant.java]:
```suggestion
    String GAMIFICATION_SOCIAL_RECEIVE_COMMENT_SPACE_STREAM                                  = "receiveCommentOnSpaceStream";
```

File: services/src/main/java/org/exoplatform/addons/gamification/listener/social/activity/GamificationActivityListener.java [hunk 7]
```diff
@@ -160,23 +160,24 @@ public void saveComment(ActivityLifeCycleEvent event) {
...
                                                            : GAMIFICATION_SOCIAL_COMMENT_NETWORK_STREAM,
                                            activityUrl);
 
     if (space != null) {
       createSpaceGamificationHistoryEntry(space.getPrettyName(),
                                           activity.getPosterId(),
-                                          GAMIFICATION_SOCIAL_COMMENT_ADD,
+                                          GAMIFICATION_SOCIAL_COMMENT_ADD_SPACE_STREAM,
                                           activityUrl);
     }
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/addons/gamification/listener/social/activity/GamificationActivityListener.java]: This should be removed, no ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/addons/gamification/listener/social/activity/GamificationActivityListener.java (hunk 7 above)
Comment (hakermi) [reply services/src/main/java/org/exoplatform/addons/gamification/listener/social/activity/GamificationActivityListener.java]: No, This gamifies the space its self, gives the space a score , used to detect popular spaces for example

**Review by rdenarie — APPROVED** [review]

**Pull request merged**